docs/linking/2-request-consent-web.puml

```diff
@@ -137,4 +141,73 @@ DFSPAuthorizeSimulator -> DFSPAuthorizeSimulator: generate a **secret="some-secr
 DFSPAuthorizeSimulator -> WebBrowser: 201 OK: response: { **secret:"some-secret"**" }
 WebBrowser -> User: redirects to callbackUri="pisp-app://callback...?**secret=some-secret**"
 deactivate User
+
+== Failure case ==
+
+autonumber 1 "<b>REQ-#</b>"
+PISP -> Switch ++: ""POST /consentRequests""\n\
+  ""  FSIOP-Source: pispa""\n\
+  ""  FSIOP-Destination: dfspa""\n\
+  "" {""\n\
+  ""  consentRequestId: "456",""\n\
+  ""  scopes: [ ""\n\
+  ""    { **accountId: "dfsp.username.1234", // This is *optional* for web.**""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""    { accountId: "dfsp.username.5678",""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""  ],""\n\
+  ""  authChannels: [ "Web", "OTP" ],""\n\
+  ""  callbackUri: "pisp-app://callback..."""\n\
+  ""}""
+
+Switch --> PISP: ""202 Accepted""
+deactivate PISP
+
+Switch -> DFSP ++:  ""POST /consentRequests""\n\
+  ""  FSIOP-Source: pispa""\n\
+  ""  FSIOP-Destination: dfspa""\n\
+  "" {""\n\
+  ""  consentRequestId: "456",""\n\
+  ""  scopes: [ ""\n\
+  ""    { accountId: "dfsp.username.1234",""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""    { accountId: "dfsp.username.5678",""\n\
+  ""      actions: [ "accounts.transfer", "accounts.getBalance" ] },""\n\
+  ""  ],""\n\
+  ""  authChannels: [ "Web", "OTP" ],""\n\
+  ""  callbackUri: "pisp-app://callback..."""\n\
+  ""}""
+DFSP --> Switch: ""202 Accepted""  
+autonumber 1 "<b>DFSP-#</b>"
+
+DFSP -> DFSP: Verify the consentRequests is valid or not?
+
+note over PISP, DFSP
+  We respond with an error that the consentRequests is invalid so the PISP can inform the user.
+end note
+
+autonumber 5 "<b>REQ-#</b>"
+DFSP -> Switch ++: ""PUT /consentRequests/456/error""\n\
+  ""  FSIOP-Source: dfspa""\n\
+  ""  FSIOP-Destination: pispa""\n\
+  ""{""\n\
+  ""  errorInformation : { ""\n\
+  ""    errorCode: "7204", ""\n\
+  ""    errorDescription: "FSP does not support any requested scope actions" ""\n\
+  ""  } ""\n\
+  ""}""
+Switch --> DFSP: ""200 OK""
+deactivate DFSP
+
+Switch -> PISP ++: ""PUT /consentRequests/456/error""\n\
+  ""  FSIOP-Source: dfspa""\n\
+  ""  FSIOP-Destination: pispa""\n\
+  ""{""\n\
+  ""  errorInformation : { ""\n\
+  ""    errorCode: "7204", ""\n\
+  ""    errorDescription: "FSP does not support any requested scope actions" ""\n\
```

Comment:
NEW ERROR CASE: `authChannels` is an array that can have only one value. So we have an error case when specified channel isn't allowed for given userId